code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardUIProvider.kt

```diff
@@ -0,0 +1,160 @@
+import com.adobe.marketing.mobile.AdobeError
+import com.adobe.marketing.mobile.Messaging
+import com.adobe.marketing.mobile.aepcomposeui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.contentprovider.AepUIContentProvider
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepUITemplate
+import com.adobe.marketing.mobile.messaging.ContentCardSchemaDataUtils.buildTemplate
+import com.adobe.marketing.mobile.messaging.ContentCardSchemaDataUtils.getAepUI
+import com.adobe.marketing.mobile.services.Log
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * ContentCardUiProvider is responsible for fetching and managing the content for a given surface.
+ * It uses Adobe Messaging APIs to retrieve propositions and transform them into UI templates for display.
+ *
+ * @property surfaceString The identifier of the surface for which content needs to be fetched.
+ */
+class ContentCardUIProvider(val surfaceString: String) : AepUIContentProvider {
+    companion object {
+        private const val SELF_TAG: String = "ContentCardUIProvider"
+    }
+
+    private val _contentFlow = MutableStateFlow<List<AepUITemplate>>(emptyList())
+    private val contentFlow: StateFlow<List<AepUITemplate>> = _contentFlow
+
+    private val _aepUiFlow = MutableStateFlow<List<AepUI<*, *>>>(emptyList())
+    private val aepUiFlow: StateFlow<List<AepUI<*, *>>> = _aepUiFlow
+
+    /**
+     * Retrieves a flow of AepUI instances for the given surface.
+     *
+     * This function initiates the content fetch using [getContent] and then returns a flow of
+     * [AepUI] instances that represent the UI templates. The flow emits updates whenever new
+     * content is fetched or any changes occur.
+     *
+     * @return A [Flow] that emits a list of [AepUI] instances.
+     */
+    suspend fun getAepUi(): Flow<List<AepUI<*, *>>> {
+        getContent()
+        return aepUiFlow
+    }
+
+    /**
+     * Clears the current content and re-fetches new content for the given surface.
+     */
+    override suspend fun refreshContent() {
+        _contentFlow.value = emptyList()
+        _aepUiFlow.value = emptyList()
+        getContent()
+    }
+
+    /**
+     * Retrieves a flow of AepUITemplate lists for the given surface.
+     * The flow emits updates whenever new content is fetched.
+     *
+     * @return A flow that emits lists of AepUITemplate.
+     */
+    override suspend fun getContent(): Flow<List<AepUITemplate>> {
+        CoroutineScope(Dispatchers.IO).launch {
+            val surface = Surface(surfaceString)
+            getAepUITemplateList(surface) { it ->
+                it.onSuccess { templateList ->
+                    _contentFlow.value = templateList
+                    _aepUiFlow.value = templateList.mapNotNull { item -> getAepUI(item) }
```

Comment:
nit: Can we move this line to `getAepUi` method? If customer is going to be using `AepList` and isn't going to be calling `getAepUi` does it make sense to maintain a flow of `aepUi`?

Reply:
I think this is being done to have `contentFlow` and `aepUiFlow` to be in sync with each other incase some one calls  `getContent()`. That makes me think if it is necessary to keep them in syc... may be we need to for maintainability. 

If we need to have them in sync-  does this work?
```
    val aepUiFlow: Flow<List<..>> = _contentFlow.map { templateList ->
        templateList.mapNotNull { item -> getAepUI(item) }
    }
```

Reply:
Changed the flow. Done